packages/react/src/ErrorBoundary.tsx

```diff
@@ -0,0 +1,52 @@
+import { Component, ErrorInfo, ReactElement, ReactNode, isValidElement } from 'react';
+
+type RenderFallback = () => ReactElement;
+
+export type Props = {
+    children: ReactNode;
+    fallback?: ReactElement | RenderFallback;
+};
+
+export type State = {
+    hasError: boolean;
+    error?: Error;
+};
```

Comment:
Can we use `interface` instead of `type` here and on `Props`?

Reply:
Sure thing. I'll fix that. Any good rule of thumb you use for deciding between type and interface?

Reply:
Fixed

Reply:
I use a type when I want to make an alias or a union, like:
```
type X = string;
type Y = string | number | OtherType | EvenOtherType;
```

and an interface when I define an object contract:
```
interface Z {
  field1: number;
  field2: boolean;
}
```

Reply:
Thanks!